source/modulo_components/include/modulo_components/ComponentInterface.hpp

```diff
@@ -1276,11 +1285,30 @@ inline void ComponentInterface<NodeT>::publish_predicates() {
   }
 }
 
+template<class NodeT>
+inline void ComponentInterface<NodeT>::publish_output(const std::string& signal_name) {
+  auto parsed_signal_name = utilities::parse_topic_name(signal_name);
+  if (this->outputs_.find(parsed_signal_name) == this->outputs_.cend()) {
+    throw exceptions::ComponentException("Output with name '" + signal_name + "' doesn't exists.");
+  }
+  if (this->periodic_outputs_.at(parsed_signal_name)) {
+    throw exceptions::ComponentException("An output that is published periodically cannot be triggered manually.");
+  }
```

Comment:
question: do you think this check is really necessary? Is it to discourage users from thinking they have to manually publish every output? I think it's ok to be restrictive but I wonder if there are cases where an extra publication during step / another place might be useful.

Reply:
It's a good question and I have thought about it as well. I kind of prefer it that way because it means we have two clearly defined options for outputs. Also, I'm not sure if we should encourage a component that needs to publish on step as well as in some other places to work correctly. That could easily lead to weird and unexpected behaviors IMO. Would you prefer to remove the check?

Reply:
I think it's good to keep the check. A bit less flexible but then more consistent / safe.